src/lib.rs

```diff
@@ -9,16 +7,16 @@ pub mod apps;
 pub mod cameras;
 pub mod deployments;
 pub mod discovery_requests;
-mod error;
+pub mod error;
 pub mod events;
 pub mod files;
 pub mod gateways;
 pub mod orgs;
 pub mod snapshots;
 pub mod streams;
 
-pub use error::HttpError;
-pub type HttpResult<T> = Result<T, HttpError>;
+use error::{verify_response, Error};
+pub type HttpResult<T> = Result<T, Error>;
```

Comment:
Since the error type is no longer called `HttpError`, the corresponding result type should be renamed too:

```suggestion
pub type Result<T, E = Error> = std::result::Result<T, E>;
```

(with an extra `E` parameter that defaults to our `Error` type so it can still be used with a different error type)